### PR TITLE
Fix storage extension build

### PR DESCRIPTION
**Pull request by JoshLove-msft**
This file was deleted in the Event Grid extension - we now use the SubscriptionValidationResponse model from the Event Grid library.

### CODE REVIEW

**Review by kasobol-msft — APPROVED** [review]

**Comment by kasobol-msft** [comment]
would be great if our build system could detect "shared source dependencies" and kick off relevant CI for them to prevent these things. not sure how often this happens though ,i.e. is it worth investment or not.

**Comment by JoshLove-msft** [comment]
> would be great if our build system could detect "shared source dependencies" and kick off relevant CI for them to prevent these things. not sure how often this happens though ,i.e. is it worth investment or not.

Yeah - agreed. I think it is not very common to use cross-service shared source (other than for Core which you would already have a dependency on). /cc @pakrym @weshaggard

**Comment by weshaggard** [comment]
Unfortunately there isn't any great way to automatically detect this by the build system and have it trigger pipelines. If that is something we want to consider I'd suggest we not share across services but move all shared stuff under Core and then if they change the core pipeline already builds all the things that depend on it.

**Comment by pakrym** [comment]
Hm, why doesn't the CI trigger?

File: sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs.csproj
Comment (kasobol-msft) [inline sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs.csproj:28]: I'm not sure we should be adding this as dependency.
We were talking the other day that it would be great to remove queues dependency and move that logic into host and expose it via some interface to extensions that need this to load balance. adding this new dependency would be a step back.
The event grid integration doesn't need whole event grid package.

@pakrym what do you think? you've been involved in these discussions.

Comment (JoshLove-msft) [reply sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs.csproj:28]: This dependency is needed to use the SubscriptionValidationResponse model. Do you have an alternative?

Comment (pakrym) [reply sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs.csproj:28]: I agree, bringing in the entire EG SDK just for this smallish feature seems heavy-handed. Can we mode the SubscriptionValidationResponse shared code into the storage SDK for now?

Comment (JoshLove-msft) [reply sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs.csproj:28]: The point of using the model from the SDK was in part to validate the scenario that customers would need to do for performing the handshake (by using an internal custom type, we didn't realize that we were missing a public ctor for the SDK model). We can ifdef this for storage but it seems correct to use the SDK model for the Event Grid extension.

Comment (kasobol-msft) [reply sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs.csproj:28]: Storage user doesn't interact with EG much. They just set the trigger source like [here](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs#event-grid-strategy) - everything else happens behind the scenes.

Comment (tg-msft) [reply sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs.csproj:28]: If this needs to be used by multiple services, we should look at moving it into Azure.Core rather than taking dependencies on EventGrid everywhere.  A shared source approach also works in the meantime.

Also, taking a dependency on EventGrid would [require explicit Arch Board approval.](https://azure.github.io/azure-sdk/general_implementation.html#general-dependencies-approved-only)

Comment (JoshLove-msft) [reply sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs.csproj:28]: We've already shipped this type as part of the Event Grid library. It is an Event Grid model - but the storage extensions use Event Grid under the hood.
I wanted to use the official Event Grid type in the extension to help ensure that the extension is doing the same thing a Webhook customer would do when using Event Grid.

Comment (tg-msft) [reply sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs.csproj]: Is Storage the only case where we'd ever want to use this?  We'd have to look at the type carefully, but we might be able to shift it into Azure.Core without breaking (i.e., introduce a base class of the EG type in Azure.Core and move all the members up to it).

Comment (JoshLove-msft) [reply sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs.csproj]: Most likely it would only be needed for Storage, but it still feels like it would be weird to put in Core since it is specific to Event Grid (and the storage extension which uses Event Grid). For now, I've shadowed the type in Storage and removed the dependency.

Comment (tg-msft) [reply sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs.csproj]: I don't think it's too weird to consider EG stuff in Azure.Core since we've already got the CloudEvent schema types and so many services use EG for surfacing notifications.  I'm happy to wait until we've got a third use for this though if we don't think it'll be common.

**Comment by JoshLove-msft** [comment]
@weshaggard it looks like this merged using automerge despite having no approvers (Kamil retracted his approval).

**Comment by weshaggard** [comment]
@JoshLove-msft that definitely looks like a bug/limitation in the auto merge feature. Probably part of the reason github still marks it as a preview. From the logs above it isn't clear the Kamil retracted his approval so I'm guessing there is something funny there.

**Comment by kasobol-msft** [comment]
@weshaggard  
image

**Comment by weshaggard** [comment]
I'm not entirely sure what "self-requested a review" means. I would have guessed that means you requested yourself to be a reviewer and not that you retracted/dismissed your existing review.

**Pull request merged**
